### ISSUE TITLE
Question: semibin_generate_data_multi doesn't produce combined data.csv 

### ISSUE THREAD

**Issue by SilasK** [BigDataBiology/SemiBin#32]
I thought the command produces one data.csv for all the samples that I feed then to the train subcommand. 
Now I got a lot of files. How should I give then to the train command? 

For each sample:

```
Crossbinning/SemiBin/samples/shRTF8.bam_0_data_cov.csv        Crossbinning/SemiBin/samples/shRTF8.bam_0_depth.txt
Crossbinning/SemiBin/samples/shRTF8.bam_0_data_split_cov.csv  Crossbinning/SemiBin/samples/shRTF8.fasta

Crossbinning/SemiBin/samples/shRTF8:
data_cov.csv  data.csv  data_split_cov.csv  data_split.csv
```

I used the command:
```
SemiBin generate_data_multi  --input-fasta Crossbinning/combined_contigs.fasta.gz  --input-bam Crossbinning/mapping/shRTF8.bam Crossbinning/mapping/shRTF2.bam Crossbinning/mapping/shHF7.bam Crossbinning/mapping/shHC1.bam Crossbinning/mapping/shRTC1.bam  --output Crossbinning/SemiBin  --threads 12  --separator _  2> log/semibin/generate_data_multi.log

```

By the way:

End of the log file:
```
/home/kiesers/scratch/Atlas/databases/conda_envs/276d10745f3e7a40761b5d1f8bc460cc/lib/python3.7/site-packages/SemiBin/generate_coverage.py:186: FutureWarning: Passing 'suffixes' which cause duplicate columns {'cov_x'} in the result is deprecated and will raise a MergeError in a future version.
  left_index=True, right_index=True, sort=False, copy=True)

```

**Comment by psj1997** [comment]
Thanks!

In our multi-sample binning, we generated data.csv for every sample but with abundance information from all samples. After that, we train model from every sample and bin every sample.

For the warning in the log file, it is about the version of pandas. I have fixed that in the latest version of SemiBin. Please use git clone and python setup.py install to use the latest SemiBin. We will update the anaconda version soon.

Sincerely
Shaojun

**Comment by SilasK** [comment]
So I should train a model for each sample to bin it individually. Does it make sense to train a model for each sample from the same project.

**Comment by psj1997** [comment]
From our results, training a model for each sample actually makes sense. For the reason that we how to handle with abundance features in multi-sample binning, we need to train a model for each sample. We will try to skip this in the future work.

**Comment by SilasK** [comment]
I'm still a bit confused. 

There is the `--mode several` for instance. 
Should the taxonomy be predicted on each sample or only on the combined.

**Comment by psj1997** [comment]
--mode several is just used in single-sample binning. In single-sample binning, SemiBin can train model from several sample. I will describe this more clear in the document.

The taxonomy is predicted on each sample.

**Comment by SilasK** [comment]
Ok, what I did to do the co-binning. Is to concatenate (and filter) the contigs from all samples. Then I created bam files by aligning all reads from all samples.

So for which step should I use the single sample fasta and for which the concatenated fasta?

Thank you very much for your help.

**Comment by psj1997** [comment]
In our tool, we classied the binning mode into single-sample, co-assembly and multi-sample binning.

For single-sample binning and coassembly binning, we used single sample fasta and bin for every sample.

For multi-sample binning,  (1) we used  concatenated fasta to generate data.csv and data_split.csv for every sample (2) we train the model and bin contigs for every sample, here we use the single sample fasta of every sample.

Sincerely
Shaojun
